themes/default/content/blog/fullstack-pulumi-mern-stack-digitalocean/index.md

```diff
@@ -0,0 +1,680 @@
+
+* A `staticSite` component mapped to the `frontend` folder
+* A `service` component mapped to the `backend` folder
+* A `database` component mapped to a managed MongoDB cluster (which we'll also configure to be accessible only by the `service` component)
+
+![Front end, back end, and database tiers mapped to their corresponding App Platform components](./mapped-tiers.png)
+
+And once deployed, it'll all be available at a single DigitalOcean-provided URL.
+
+Let's begin by creating new Pulumi project.
+
+## Creating the project
+
+In the root of the repository, make a new folder called `infra`, change to it, then run `pulumi new` using the `digitalocean` [project template](https://github.com/pulumi/templates):
+
+{{% chooser language "typescript,python" /%}}
+
+{{% choosable language typescript %}}
+
+```bash
+$ mkdir infra && cd infra
+$ pulumi new digitalocean-typescript
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```bash
+$ mkdir infra && cd infra
+$ pulumi new digitalocean-python
+```
+
+{{% /choosable %}}
+
+At the prompts, use the following values:
+
+* For project name, use `grocery-list`
+* For description, use `Deploying a MERN-stack app on DigitalOcean`
+* For stack name, use `dev`, the default
+
+When the command completes, you'll have a new [Pulumi stack]({{< relref "/intro/concepts/stack" >}}), but you'll still have a few things to configure. For one, the Pulumi DigitalOcean provider needs to be [configured]({{< relref "/registry/packages/digitalocean/installation-configuration" >}}) to communicate with DigitalOcean on your behalf (to provision your app and its various resources). For this, you can use the access token you obtained [earlier](#setting-up) from the DigitalOcean console, and you can apply it by setting a single environment variable:
+
+```bash
+$ export DIGITALOCEAN_TOKEN="your-access-token"
+```
+
+Also, since one of our goals is to have App Platform deploy automatically on every GitHub commit, you'll need to tell DigitalOcean where to find the source code for your front- and back-end components. You _could_ bake these settings right into the Pulumi program itself, but it'd be better to apply them as stack-specific configuration settings, as that'd let you deploy to different stacks later (say, in CI) based on the branch of the commit. Everyone does this a little differently, so for now, let's configure the currently selected stack (which should be `dev`) to use the default branch of your GitHub repository:
```

Comment:
Question for the future (maybe a future update to the post?): Can this be done with GitLab as well? I'm betting someone will ask.

Reply:
It can! I'll add a note for that as well.